projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -96,6 +78,12 @@ public function should_disable_links_manager() {
 	 * Adds My Home menu.
 	 */
 	public function add_my_home_menu() {
+		// When no preferred view has been set for "My Home", keep the previous behavior that forced the default view
```

Comment:
```suggestion
		// Strictly check if CLASSIC_VIEW is selected, rather than depending on should_link_to_wp_admin() for users who haven't made a selection yet.
```

Reply:
I'll change the implementation to better reflect the current comment:
```
// When no preferred view has been set for "My Home", keep the previous behavior that forced the default view
// regardless of the global preference.
$preferred_view = $this->get_preferred_view( 'index.php', true );
if ( self::DEFAULT_VIEW === $preferred_view || self::UNKNOWN_VIEW === $preferred_view ) {
	$this->update_menu( 'index.php', 'https://wordpress.com/home/' . $this->domain, __( 'My Home', 'jetpack' ), 'manage_options', 'dashicons-admin-home' );
}	
```

Reply:
Not needed after moving all the exceptions to the `get_preferred_view` function in https://github.com/Automattic/jetpack/pull/20147/commits/394c0241611abf41b6e8ba838cac04df7883b904